src/vue/components/inflections.vue

```diff
@@ -151,25 +129,28 @@ export default {
 
   computed: {
     isEnabled: function () {
-      return this.data.inflectionViewSet && this.data.inflectionViewSet.enabled
+      return this.$store.state.app.inflectionsViewSet && this.$store.state.app.inflectionsViewSet.enabled
     },
     hasMatchingViews: function () {
-      return this.data.inflectionViewSet && this.data.inflectionViewSet.enabled && this.data.inflectionViewSet.hasMatchingViews
+      return this.$store.state.app.inflectionsViewSet && this.$store.state.app.inflectionsViewSet.enabled && this.$store.state.app.inflectionsViewSet.hasMatchingViews
     },
     inflectionViewSet: function () {
-      return this.data.inflectionViewSet
+      return this.$store.state.app.inflectionsViewSet
     },
     // Need this for a watcher that will monitor a parent container visibility state
     isVisible: function () {
-      return this.data.visible
+      return this.$store.state.app.tabState.inflections
+    },
+    inflectionsEnabled: function () {
+      return ViewSetFactory.hasInflectionsEnabled(this.$store.state.app.currentLanguageID)
```

Comment:
I think such tests, especially if they are not component-specific, should belong to a UI controller. That would be in line with our decision to move business logic into if.

This thing here is just temporary. I did not have time to fully analyze and refactor all state checks in inflections: there are so many of them in different places like visible/enabled/hasData/hasMatchingViews etc., it's very messy at the moment on my opinion; we should make them more transparent. But it's tricky because one little change can break things in so many places.

This piece of code happened to be there because during reduction of functions this seemed to be a logical place to have it without breaking too many other relationships.

I think I should probably move it to a UI controller. What is your opinion on that?

Reply:
I don't really know. I think you could make an argument either way. For example, why should the UIController need to know that the inflections tab or grammar tab needs to be reset for a new query? If that is always the desired behavior, then this and the grammar reset too probably belong in the respective components. On the other hand, maybe that's not true. Since the UI Controller is responsible for coordination the flow of actions across the interface, and actually initiating the query which is driving it, then it is possible that a different application might actually have different requirements here and wish for the inflections tab or grammar tab to stay populated. 

I think the most important thing right now is to try to achieve a logical consistency, and I definitely understand that we need to do this refactoring incrementally. So when we come across things like this I think if we leave anything for that reason, i.e. because we don't want to break something yet, then it would be a good idea to put a comment in the code to that effect so that we can easily see why the inconsistency is there. Does that seem reasonable?

Reply:
Yes, absolutely. Added a TODO.